License2Deploy/rolling_deploy.py

```diff
@@ -319,12 +333,30 @@ def enable_project_cloudwatch_alarms(self):
         logging.error("Unable to enable the cloud-watch alarm, please investigate: {0}".format(e))
         exit(self.exit_error_code)
 
+  def is_redeploy(self):
+    current_reservations = self.get_reservations(self.existing_instance_ids)
+    current_build_numbers = [int(instance.tags['BUILD'])
+                             for reservation in current_reservations
+                             for instance in reservation.instances
+                             if 'BUILD' in instance.tags]
+    if not current_build_numbers:
+      self.stop_deploy('Failed to determine current build. Ensure instances contain tag "BUILD"')
+    return int(self.build_number) in current_build_numbers
+
+  def stop_deploy(self, message='an error has occurred', e=None, error_code=2):
+    logging.error('{0}: {1}'.format(message, e))
+    exit(error_code)
+
   def deploy(self): # pragma: no cover
     self.load_balancer = self.get_lb()
     ''' Rollin Rollin Rollin, Rawhide! '''
     group_name = self.get_autoscale_group_name()
     self.wait_ami_availability(self.ami_id)
     logging.info("Build #: {0} ::: Autoscale Group: {1}".format(self.build_number, group_name))
+    self.existing_instance_ids = list(self.get_all_instance_ids(group_name))
+    self.log_instances_ips(self.existing_instance_ids, group_name)
+    if self.is_redeploy() and not self.force_redeploy:
```

Comment:
I suggest switching the logic here so that it won't make the boto call on force_redeploy
```if not self.force_redeploy and self.is_redeploy():```